### PR TITLE
JSDoc documentation generation

### PR DESCRIPTION
**Pull request by inomn**
[Documentation is here](https://goodhood-eu.github.io/nebenan-react-hocs/index.html)

[Favro](https://favro.com/organization/aee06c3fe6503c804f507dd9/22c630cc02b09f2d6f98b8b3?card=Goo-42825)

### CODE REVIEW

File: src/proxy_connect/index.jsx [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import { connect } from 'react-redux';
 import { getDisplayName, getForwardedComponent } from '../utils';
 
-
+/**
+ * @param mapStateToProps
+ * @param mapDispatchToProps
+ * @param mergeProps
+ * @param opts
+ * @return { React.Component } Wrapped with Redux.Connect component
```
Comment (peterhass) [inline src/proxy_connect/index.jsx:9]: does it matter if there is a space after and before the brackets?

File: src/redux_hooks/index.jsx [hunk 2]
```diff
@@ -3,7 +3,11 @@ import { useMemo } from 'react';
 import { bindActionCreators as bind } from 'redux';
 import { useSelector, shallowEqual, useDispatch } from 'react-redux';
 
-
+/**
+ * @param actions
+ * @param deps
+ * @return {unknown}
```
Comment (peterhass) [inline src/redux_hooks/index.jsx]: returns either array of functions or one single function

File: src/use_debounced_callback/index.jsx [hunk 3]
```diff
@@ -1,7 +1,13 @@
 import debounce from 'lodash/debounce';
 import { useEffect, useRef } from 'react';
 
-// Does not react to `delay` changes
+/**
+ * Note: Does not react to `delay` changes
+ * @param func
+ * @param delay
+ * @param deps
+ * @return {null}
```
Comment (peterhass) [inline src/use_debounced_callback/index.jsx]: returns function?

File: src/use_event_listener/index.jsx [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
 
+/**
+ * @param ref
+ * @param type
+ * @param handler
+ * @param passive
+ * @param capture
+ * @param once
+ */
 export const useEventListener = (
```
Comment (peterhass) [inline src/use_event_listener/index.jsx:12]: any idea why it doesn't get detected as a function? 

https://goodhood-eu.github.io/nebenan-react-hocs/global.html#useEventListener

**Review by peterhass — CHANGES_REQUESTED** [review]
Really nice to see this thing coming to life!

https://goodhood-eu.github.io/nebenan-react-hocs/global.html

The only reason why I am requesting changes is because of the 'Members'  section. There we don't see any params. Why don't they show up under 'Methods'?

File: src/proxy_connect/index.jsx (hunk 1 above)
Comment (inomn) [reply src/proxy_connect/index.jsx:9]: nope :( , on generated page it will be without spaces anyway

File: src/use_event_listener/index.jsx (hunk 4 above)
Comment (inomn) [reply src/use_event_listener/index.jsx:12]: ES6 \ classes should be declared with `@function` (done)

File: src/adaptive/index.jsx [hunk 5]
```diff
@@ -3,7 +3,12 @@ import eventproxy from 'nebenan-helpers/lib/eventproxy';
 import { getMedia, media } from 'nebenan-helpers/lib/dom';
 import { getDisplayName, getForwardedComponent } from '../utils';
 
-
+/**
+ * @function useIsMobile
+ * @description Subscription on Resize
+ * @description See nebenan-helpers/lib/dom
```
Comment (peterhass) [inline src/adaptive/index.jsx]: merge into one - jsdoc only shows one @description block

File: src/adaptive/index.jsx [hunk 6]
```diff
@@ -14,6 +19,11 @@ export const useIsMobile = () => {
   return isMobile;
 };
 
+/**
+ * @param {React.Component} Component Target component
```
Comment (peterhass) [inline src/adaptive/index.jsx:22]:
```suggestion
 * @param {React.Component} Target component
```

**Review by peterhass — APPROVED** [review]

File: src/adaptive/index.jsx (hunk 6 above)
Comment (inomn) [reply src/adaptive/index.jsx:22]: {type} FN_PropName PropDescription
Screenshot 2021-10-07 at 16 21 20


in IDE:
Screenshot 2021-10-07 at 16 21 39

**Pull request merged**